third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -1222,10 +1265,15 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
         let fields = match &def.fields {
             EA::StructFields::Defined(fields) => {
                 let mut field_map = BTreeMap::new();
+                let mut field_ty_and_locs = Vec::new(); // fix borrowing issues
                 for (name_loc, field_name_, (idx, ty)) in fields {
                     let field_loc = et.to_loc(&name_loc);
                     let field_sym = et.symbol_pool().make(field_name_);
                     let field_ty = et.translate_type(ty);
+                    let field_ty_loc = et.to_loc(&ty.loc);
+                    field_ty_and_locs.push((field_ty.clone(), field_ty_loc));
+                    // borrow checker complains
+                    // self.ability_check_field(struct_abilities, &qsym, &field_ty, &field_ty_loc);
```

Comment:
I think including a commented out line of code is confusing.  Why not say, about the previous line: 
```
// store the `field_ty` and f`ield_ty_loc `to process with `ability_check_field` outside of this loop, to avoid borrow issues
field_ty_and_locs.push((field_ty.clone(), field_ty_loc));
```
Although I wonder if you could do the job here if you also cloned `field_ty`:
```
self.ability_check_field(struct_abilities, &qsym, &(field_ty.clone()), @field_ty_loc);
```
or is something else responsible for the borrow issue?

Reply:
The borrow issue is caused by the `et` which borrows `self` mutably. We can solve the issue by using a new et in each iteration, but the overhead is larger than the current approach (though perhaps easier to read).

Reply:
done